project/src/main/kotlin/org/ionproject/core/common/Uri.kt

```diff
@@ -5,14 +5,11 @@ import org.springframework.web.util.UriTemplate
 import java.net.URI
 
 object Uri {
+    const val baseUrl = "https://host1.dev.ionproject.org"
```

Comment:
This should not be a hardcoded string. A different build would be needed for running a local instance and everytime the remote host changed. 

A solution might to use the host component of the HTTP request received. This can be done by having a parameter of type `WebRequest` as shown [here](https://docs.spring.io/spring/docs/current/spring-framework-reference/web.html#mvc-ann-arguments).

Reply:
The current use of this string is only to generate import url's, import url's will never be used localhost. And what do you mean by HTTP request received im not understanding the second part?

Reply:
I know it is for imports. But if I ask a server for an import URL I expect it to reference the same server I made the request to.

The `WebRequest` class allows you to know what host was used in the request so you don't need it hardcoded.

Also I don't think it makes sense the need to have a different build of the server if the domain changes.

Reply:
As discussed in private chat clearly the disvantages are too big for the current strategy, and was also decided for security problems that usigng the header value was an insecure option because of possible malicious user input. Therefore was used as agreed HttpServletRequest.GetServerName.

Reply:
Update: I just checked the implementation of getservername, getrequesturl... and they use the value of the header host. Which makes this solution invalid.

Reply:
Update2: Important to clarify, that this is not really a "problem" at this phase, and only if the API evolves to have different hostnames for different resources, a person could bypass that validation or a distracted developer ends up using the value not knowing its unsafe.

Reply:
It was decided in the meeting of July 6 2020, to use environment variables.